test/support/index.js

```diff
@@ -90,27 +91,55 @@ var Support = {
   },
 
   clearDatabase: function (sequelize, callback) {
-    sequelize
-      .getQueryInterface()
-      .dropAllTables()
-      .then(function () {
-        if (sequelize.daoFactoryManager) {
-          sequelize.daoFactoryManager.daos = [];
-        } else {
-          sequelize.modelManager.models = [];
-        }
-
-        return sequelize
-          .getQueryInterface()
-          .dropAllEnums()
-            .then(callback)
-            .catch(function (err) {
-              console.log('Error in support.clearDatabase() dropAllEnums() :: ', err);
+    function dropAllTables() {
+      return sequelize
+        .getQueryInterface()
+        .dropAllTables()
+        .then(function () {
+          if (sequelize.daoFactoryManager) {
+            sequelize.daoFactoryManager.daos = [];
+          } else {
+            sequelize.modelManager.models = [];
+          }
+
+          return sequelize
+            .getQueryInterface()
+            .dropAllEnums()
+              .then(callback)
+              .catch(function (err) {
+                console.log('Error in support.clearDatabase() dropAllEnums() :: ', err);
+              });
+        })
+        .catch(function (err) {
+          console.log('Error in support.clearDatabase() dropAllTables() :: ', err);
+        });
+    };
+
+    // If Postgres, loop through each of the non-public schemas and DROP/re-CREATE them.
+    if (this.dialectIsPostgres()) {
+      sequelize
+        .showAllSchemas()
+        .then(function (schemas) {
+          // showAllSchemas() leaves off the public schema.
+          let searchPathSchemas = schemas.join(', ');
+          let promise = Bluebird.resolve();
+
+          _.forEach(schemas, function (schema) {
```

Comment:
Use `Promise.map` + `Promise.all` combination rather than `forEach` for better readability